simulation/groups.py

```diff
@@ -6,337 +6,205 @@
+        Area around IGroup instance in which SGroup instance could be infected.
+        It is a radius value.
+    recover_prob: float, required
+        Probability to trigger the recover behaviour it should be value
+        between 0.0 and 1.0.
+    death_prob: float, required
+        Probability trigger the death behaviour it should be value
+        between 0.0 and 1.0.
+    prefix: str, optional
+        Prefix for auto labeling logic which is {prefix}_{id}.
+        Should not be changed. Default value: infected.
+    """
+
+    id = count(1)
+
+    def __init__(self, x: float, y: float, infection_range: float,
+                 recover_prob: float, death_prob: float, prefix: str = 'infected') -> None:
+        self.recover_probability = recover_prob
+        self.death_probability = death_prob
+        self.infection_range = infection_range
+        super().__init__(x, y, prefix)
+
+    def recover(self) -> bool:
+        """
+        Method checks if current instance get recovered based on attribute
+        recover_probability.
+
+        Returns
+        -------
+        True if instance get recovered or False if not. Logic base on
+        random.choices() method.
+
+        Example
+        -------
+        >>> i_member.recover()
+        """
+
+        return super().behaviour_probability(self.recover_probability)
+
+    def death(self) -> bool:
+        """
+        Method checks if current instance died based on attribute
+        death_probability.
+
+        Returns
+        -------
+        True if instance died or False if not. Logic base on
+        random.choices() method.
+
+        Example
+        -------
+        >>> i_member.death()
+        """
+
+        return super().behaviour_probability(self.death_probability)
 
 
 class RGroup(groups_interface.PopulationGroup):
-    def __init__(self, x: float, y: float, label: str, inf_dist: float, infection_prob: float = 0,
-                 recover_prob: float = 0, death_prob: float = 0) -> None:
-
-        super().__init__(x, y, label, inf_dist, infection_prob, recover_prob, death_prob)
-
-    def __del__(self):
-        pass
-
-    def move(self, x_distance: float, y_distance: float, box_width: int,
-             box_height: int) -> bool:
+    """
+    Class represent people that can not be infected second time. Instances of
+    this class do not define specific behaviours.
 
-        can_move = True
+    Parameters
+    ----------
+    x: float, required
+        Instance starting position on X axis on cartesian grid defined by
+        container wrapper.
+    y: float, required
+        Instance starting position on Y axis on cartesian grid defined by
+        container wrapper.
+    prefix: str, optional
+        Prefix for auto labeling logic which is {prefix}_{id}.
+        Should not be changed. Default value: recovered.
+    """
 
-        if not 0 < self.x + x_distance < box_width:
-            can_move = False
+    id = count(1)
```

Comment:
Be careful with class variables in Python (class variables are shared between class instances / objects). Here, every instance of RGroup will have this `id`.